UnitTests/GitExtUtils/GitArgumentBuilderExtensionsTest.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Linq;
+using GitCommands;
+using NUnit.Framework;
+
+namespace GitExtUtilsTests
+{
+    [TestFixture]
+    public class GitArgumentBuilderExtensionsTest
+    {
+        // 17: 'git checkout name1'
+        [TestCase("checkout", new string[] { "name1", "name2", "name3" }, 21, new string[] { "checkout name1", "checkout name2", "checkout name3" })]
+        [TestCase("checkout", new string[] { "name1", "name2", "name3" }, 22, new string[] { "checkout name1", "checkout name2", "checkout name3" })]
+        [TestCase("checkout", new string[] { "name1", "name2", "name3" }, 27, new string[] { "checkout name1 name2", "checkout name3" })]
```

Comment:
I would prefer to test the exact thresholds.

Reply:
Test case for exact thresholds are added in `ExecutableExtensionsTests` fixture, which will call the actual `Process`. This only performs unit testing for the builder extensions.